### PR TITLE
Add '--test-only' flag in perf test

### PR DESCRIPTION
**Pull request by eunwoosh**
### Summary
This PR adds `--test-only` flags to perf test. When it's set, perf test copies all necessary files from resume directory and executes only `otx test` with specified operation. If files can't be found from resume directory, necessary operations will be executed automatically.
Additionally, bug that error is raised when setting device to `cpu` is fixed.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by goodsong81 — COMMENTED** [review]
Isn't it a little bit misleading not executing `otx export` when specifying `--resume-from export`? It sounds like rather `--resume-after` to me.
(Not saying that it should be renamed to "after", thouth)

**Comment by eunwoosh** [comment]
> Isn't it a little bit misleading not executing `otx export` when specifying `--resume-from export`? It sounds like rather `--resume-after` to me. (Not saying that it should be renamed to "after", thouth)

you're right, but it's hard to find proper name. So far, `--resume-from` is best candidate, I think. I'll think more about that.

**Review by kprokofi — COMMENTED** [review]
I don't understand, why should we add second option with resume_from. Why only one resume is not enough to continue training? Let's imagine we have some workspace with already completed train stage for small datasets. Can we just write: pytest test_*.py --work-dir /path/to/it --eval-upto optimize --resume. After that I would expect that export -> optimize on small datasets will be executed

It looks complicated for me with two resumes

**Comment by eunwoosh** [comment]
> I don't understand, why should we add second option with resume_from. Why only one resume is not enough to continue training? Let's imagine we have some workspace with already completed train stage for small datasets. Can we just write: pytest test_*.py --work-dir /path/to/it --eval-upto optimize --resume. After that I would expect that export -> optimize on small datasets will be executed
> 
> It looks complicated for me with two resumes

It's for saving more time when running huge number of models. Current perf test resume feature just skips training part. So, if user uses `--resume-from`, then perf test starts from `otx test` after training. Of course, skipping training part can save a lot of time, but sometimes skipping additional parts is preferred. For examples, recently I tried to benchmark score using all OTX models' optimized OV IR. With current resume feature, I need to run `otx test` 15 times using trained model, OV IR and quantized OV IR for every model (repeat 5 x 3 `otx test`). In this case, skipping `otx test` w/ trained model and OV IR can save a lot of time. And it can be beneficial when using environment w/o any acclerator where `otx test` w/ trained model spends lots of time.

**Comment by goodsong81** [comment]
> > I don't understand, why should we add second option with resume_from. Why only one resume is not enough to continue training? Let's imagine we have some workspace with already completed train stage for small datasets. Can we just write: pytest test_*.py --work-dir /path/to/it --eval-upto optimize --resume. After that I would expect that export -> optimize on small datasets will be executed
> > It looks complicated for me with two resumes
> 
> It's for saving more time when running huge number of models. Current perf test resume feature just skips training part. So, if user uses `--resume-from`, then perf test starts from `otx test` after training. Of course, skipping training part can save a lot of time, but sometimes skipping additional parts is preferred. For examples, recently I tried to benchmark score using all OTX models' optimized OV IR. With current resume feature, I need to run `otx test` 15 times using trained model, OV IR and quantized OV IR for every model (repeat 5 x 3 `otx test`). In this case, skipping `otx test` w/ trained model and OV IR can save a lot of time. And it can be beneficial when using environment w/o any acclerator where `otx test` w/ trained model spends lots of time.

I think @kprokofi's comment makes sense. Isn't it some kind of test-only feature that you really need?
Why don't you add some flag like "--test train|export|optimize|all"?

**Comment by eunwoosh** [comment]
I understand what you said. I'll revert "--resume" to "--resume-from" and change "--resume-from" to "--test-only".

File: tests/perf/benchmark.py [hunk 1]
```diff
@@ -128,23 +136,26 @@ def __init__(
         self.deterministic = deterministic
         self.accelerator = accelerator
         self.reference_results = reference_results
+        self.resume_from = resume_from
+        if (test_only == "export" and eval_upto == "train") or (
```
Comment (sungchul1) [inline tests/perf/benchmark.py:140]: What do you think about using enum instead of comparing strings?

Comment (eunwoosh) [reply tests/perf/benchmark.py:140]: It's a fine idea but there is some ambiguous point there. `eval_upto` is expected to get `train`, `export` or `optimize` but `resume_from` can also get `all` including them. So, even if there is enum class, using it for both of them looks a little bit weird. Also comparing between values including `all` is also other ambiguous point. For these reasons, I think it's better to leave it as it is for now.

**Review by goodsong81 — APPROVED** [review]
LGTM

**Review by sungchul1 — APPROVED** [review]

**Pull request merged**